src/astro/sql/operators/sql_dataframe.py

```diff
@@ -1,35 +1,68 @@
 import inspect
-from typing import Dict, Optional
+from typing import Callable, Dict, Optional, Tuple
 
 import pandas as pd
 from airflow.decorators.base import DecoratedOperator
-from airflow.providers.sqlite.hooks.sqlite import SqliteHook
 
-from astro.constants import Database
 from astro.databases import create_database
 from astro.settings import SCHEMA
 from astro.sql.table import Table
-from astro.sqlite_utils import create_sqlalchemy_engine_with_sqlite
 from astro.utils import get_hook
-from astro.utils.database import create_database_from_conn_id
-from astro.utils.dependencies import (
-    BigQueryHook,
-    PostgresHook,
-    SnowflakeHook,
-    postgres_sql,
-)
+from astro.utils.dependencies import SnowflakeHook
 from astro.utils.load import load_dataframe_into_sql_table
 from astro.utils.table_handler import TableHandler
 
 
+def _get_dataframe(table: Table, identifiers_as_lower: bool = False) -> pd.DataFrame:
+    """
+    grabs a SQL table and converts it into a dataframe
+    :param table:
+    :return:
+    """
+    database = create_database(table.conn_id)
+    df = database.export_table_to_pandas_dataframe(source_table=table)
+    if identifiers_as_lower:
+        df.columns = [col_label.lower() for col_label in df.columns]
+    return df
+
+
+def load_op_arg_table_into_dataframe(
+    op_args: Tuple, python_callable: Callable
+) -> Tuple:
+    """For dataframe based functions, takes any Table objects from the op_args
+    and converts them into local dataframes that can be handled in the python context"""
+    full_spec = inspect.getfullargspec(python_callable)
+    op_args_list = list(op_args)
+    ret_args = []
+    for arg in op_args_list:
+        current_arg = full_spec.args.pop(0)
+        if full_spec.annotations[current_arg] == pd.DataFrame and type(arg) is Table:
```

Comment:
Why are we checking annotation? What if user does not add type annotation?

Reply:
```suggestion
        if isinstance(current_arg, pd.DataFrame) and isinstance(arg, Table):
```
might work?

Reply:
@kaxil without the type annotation there's no way for us to know what something is supposed to be.

Reply:
@kaxil that suggestion won't work. current_arg is a string.

Reply:
The following links give good summary of why `isinstance` should be preferred instead of `type`. tl_dr: to cater for inheritance.

- https://stackoverflow.com/questions/1549801/what-are-the-differences-between-type-and-isinstance
- https://stackoverflow.com/questions/30049410/why-use-isinstance-instead-of-type

Reply:
@kaxil I can change the second part to isinstance but I am fairly certain that trying to move away from the annotation would send me down a rabbit hole. If you want to try it we can make it a separate PR.